include/pa_oboe.h

```diff
@@ -0,0 +1,154 @@
+    Unprocessed = 9 , VoicePerformance = 10
+};
+
+/**
+ * Enum class that emulates Oboe::PerformanceMode
+ */
+enum class PaOboe_PerformanceMode : int32_t { None = 10 , PowerSaving = 11 , LowLatency = 12 };
+
+/**
+ * Enum class that emulates Oboe::SharingMode
+ */
+enum class PaOboe_SharingMode : int32_t {  Exclusive = 0, Shared = 1 };
+
+/**
+ * Enum class that emulates Oboe::ContentType
+ */
+enum class PaOboe_ContentType : int32_t { Speech = 1, Music = 2, Movie = 3, Sonification = 4 };
+
+/**
+ *  The android stream type and recording preset as defined in Oboe.
+ */
+typedef struct PaOboeStreamInfo {
+    // REquired fields to allow casting into PaUtilHostApiSpecificStreamInfoHeader
+    unsigned long size;
+    PaHostApiTypeId hostApiType; 
+    unsigned long version;
+
+    PaOboe_Usage androidOutputUsage;
+    PaOboe_InputPreset androidInputPreset;
+    PaOboe_PerformanceMode performanceMode;
+    PaOboe_SharingMode sharingMode;
+    PaOboe_ContentType contentType;
+    const char* packageName;
+} PaOboeStreamInfo;
+
+/** Initialize host API specific structure, call this before setting relevant attributes. */
+void PaOboe_InitializeStreamInfo( PaOboeStreamInfo *info );
+
+/**
+ * Provide PA Oboe with native buffer information. If you call this function, you must do so before
+ * calling Pa_Initialize. To have optimal latency, this function should be called - otherwise,
+ * PA Oboe will use potentially non-optimal values as default.
+ * @param bufferSize the native buffersize as returned by AudioManager's
+ * PROPERTY_OUTPUT_FRAMES_PER_BUFFER. It is recommended you set the number of buffers to 1 if API>17
+ * as well, and use the sample rate defined in AudioManager's android.media.property.OUTPUT_SAMPLE_RATE.
+ * All three together will enable the AUDIO_OUTPUT_FLAG_FAST flag.
+ */
+void PaOboe_SetNativeBufferSize(unsigned long bufferSize);
+
+/**
+ * Provide PA Oboe with native buffer information. If you call this function, you must do so before
+ * calling Pa_Initialize. To have optimal latency and enable the AUDIO_OUTPUT_FLAG_FAST flag, this
+ * function should be called - otherwise, PA Oboe will use potentially non-optimal values (2) as default.
+ * @param buffers The number of buffers can be reduced to 1 on API >17. Make sure you set the native
+ * buffer size when doing this, and use the sample rate defined in AudioManager's
+ * android.media.property.OUTPUT_SAMPLE_RATE.
+ */
+void PaOboe_SetNumberOfBuffers(unsigned numberOfBuffers);
+
+/**
+ * Since Obee doesn't have the ability to detect devices, this helper functions allows the user to register devices, 
+ * previously detected thanks to the JNI interface, since Google is committed to only allow this option to detect devices.
+ * @param name The device name
+ * @param direction The device direction
+ * @param channelCounts The device channel count
+ * @param sampleRate The device default sampleRate
+ */
```

Comment:
It seems like we would also need a deviceId so that we can select the device using builder.setDeviceId(id). Maybe I am missing something.